### PR TITLE
chore(validation): [OSL-312] Remove emoji from list slugs

### PR DESCRIPTION
**Pull request by nina-py**
## Goal

Improve slugs for shared lists.

- Updated `slugify` config to remove all characters that don't exist in the package's char map: https://github.com/simov/slugify/blob/master/config/charmap.json

- Added a neutral slug text for cases when list title is composed entirely of emoji or other characters that would be removed during the slugifying process: `shared-list`.

- Added two new integration tests to cover updated functionality.

## Tickets

https://getpocket.atlassian.net/browse/OSL-312

### CODE REVIEW

**Review by jpetto — CHANGES_REQUESTED** [review]
this is a great solution! i think we need one small change and one more test and it'll be good to go.

File: src/database/mutations/ShareableList.ts [hunk 1]
```diff
@@ -138,23 +138,30 @@ export async function updateShareableList(
...
         userId,
-        slug: { contains: slugify(data.title ?? list.title, config.slugify) },
+        slug: { contains: slugifiedTitle },
       },
     });
+
+    // if title was made up entirely of characters that the slug cannot contain,
+    // e.g. emojis, generate a neutral-sounding, short slug
+    const preparedSlug =
+      slugifiedTitle.length > 0 ? slugifiedTitle : 'shared-list';
```
[the remark this answers is not in the document]
Comment (nina-py) [reply src/database/mutations/ShareableList.ts:155]: Good catch, thank you, Jonathan!

File: src/database/mutations/ShareableList.ts [hunk 2]
```diff
@@ -141,19 +141,19 @@ export async function updateShareableList(
...
+
     // First check how many slugs containing the list title already exist in the db
     const slugCount = await db.list.count({
       where: {
         userId,
-        slug: { contains: slugifiedTitle },
+        slug: { contains: preparedSlug },
```
Comment (jpetto) [inline src/database/mutations/ShareableList.ts:153]: i know it's outside the scope of this PR, but i wonder if `startsWith` would be better here... definitely not a blocker.

**Review by jpetto — APPROVED** [review]
looks great! ship it!

**Pull request merged**